include/oboe/AudioStream.h

```diff
@@ -544,6 +542,27 @@ class AudioStream : public AudioStreamBase {
         return mPerformanceHintEnabled;
     }
 
+    /**
+     * If you have called setPerformanceHintEnabled(true) then you can give
+     * the performance manager more information about your workload.
+     * You can call this at the beginning of the callback when you figure
+     * out what your workload will be.
+     *
+     * The workload can be in your own units. If you are synthesizing music
+     * then the workload could be the number of active voices.
+     * If your app is a game then it could be the number of sound effects.
+     *
+     * The performance hint code can use this as an advance warning that the callback duration
```

Comment:
How many different sources of truth will we have for the performance hint after this CL? What if multiple are actively used?

How would apps know that they are doing the right thing when they call this function? Is there some easy graphs or traces we can point them to?

Reply:
ADPF works on a per-thread basis. Then the scheduler balances the requests from all the threads.

It is a bit fuzzy and indirect. If the glitches stop then it is probably working. They could also look at Perfetto traces.